### ISSUE TITLE
Backslash in URLs breaks website in Firefox and IE

### ISSUE THREAD

**Issue by kenlyon** [justinwalsh/daux.io#264]
I downloaded daux.io-master and ran "npm install" followed by "grunt" without changing any of the default settings or content. The site shows up correctly in Chrome but expand/collapse links such as "Examples" and "More Examples" don't work in IE and there is no style at all in Firefox.

Viewing the source, I see links like this:

&lt;link href='//127.0.0.1:8085\/templates/default/themes/daux-blue/css/daux-blue.css' rel='stylesheet' type='text/css'&gt;

I don't know why there's a backslash after the port number, but that almost certainly is the cause of the problem. (All three browsers show this in the source, but they have mixed results in handling it.)

My environment:
Windows 7 (64-bit) SP1
node.js 0.12.2
php 5.6.0
Firefox 37.0.1

**Comment by theiwaz** [comment]
I've had the same problem. The problem is in `daux.php` on line 80. There it says
```php
$this->base_url = $_SERVER['HTTP_HOST'] . dirname($_SERVER['PHP_SELF']) ;
```
The issue on windows arrises from `dirname`, it returns a backslash instead of a forward slash as the directory seperators. I've replaced it with:
```php
$this->base_url = $_SERVER['HTTP_HOST'] . str_replace(dirname($_SERVER['PHP_SELF']),"\\","/");
```

**Comment by kenlyon** [comment]
Thanks for fixing that, @theiwaz! My website now looks great in Firefox, IE, Chrome and Safari with no backslashes in sight.

I'm not sure of the best process for how to proceed. I don't want to close this until your change is accepted.

**Comment by kminek** [comment]
i can confirm that this change fixes problems with daux on windows (i am using xampp). without this change URLs are broken

**Comment by onigoetz** [comment]
Can you please try with the latest development branch ? we made a big refactoring and would love to know if it works for that case.

I will close your issue as it is old, feel free to reopen it if the problem persists

**Comment by kenlyon** [comment]
I downloaded the development zip and retested this. It's still an issue for me. I still see the backslash in the URLs. I tested in Firefox and Chrome, clearing my cache as well to ensure that's not a factor. I still see URLs like this:

//127.0.0.1:8085\/resources/themes/daux-blue/css/theme.min.css

Could you please re-open this? I don't seem to have permission to do so.

**Comment by onigoetz** [comment]
I will try to get a windows machine running and try this, The issue is now reopened

**Comment by kenlyon** [comment]
Thanks!

**Comment by onigoetz** [comment]
Hi, this is fixed in the latest development fixes

**Comment by kenlyon** [comment]
Do you have any update on this? Is there an earlier commit in develop that I can use to verify this fix? So far it's still broken as far as I can tell.

**Comment by onigoetz** [comment]
Hello, I will try to reproduce it with grunt, 

what happens when you run the php server by hand with `php -S 0.0.0.0:8085 index.php` ?

I'm sorry I'm all out of ideas, it works correctly for me on windows 7 with wamp and with the development server

**Comment by kenlyon** [comment]
I tried your suggestion and got exactly the same experience as before.

* Downloaded and unzipped the latest daux.io-development.zip.
* Opened a command prompt and ran "npm install"
* Executed the command "php -S 0.0.0.0:8085 index.php"
* Opened http://127.0.0.1:8085/ in my browser.

The page was completely empty, with no errors in the browser. Viewing the source revealed it was a completely empty page. I did see the following appear in the command prompt window, though:

PHP 5.6.0 Development Server started at Tue Sep 08 08:57:40 2015
Listening on http://0.0.0.0:8085
Document root is C:\Users\\********\Downloads\daux.io-development
Press Ctrl-C to quit.
[Tue Sep 08 08:58:08 2015] 127.0.0.1:63109 Invalid request (Unexpected EOF)

Subsequent requests did not add more "Invalid request" messages to the output, though.
